src/System.IdentityModel.Tokens.Jwt/Validators.cs

```diff
@@ -169,8 +169,14 @@ public static void ValidateIssuerSecurityKey(SecurityKey securityKey, SecurityTo
             X509SecurityKey x509SecurityKey = securityKey as X509SecurityKey;
             if (x509SecurityKey != null)
             {
-                //validationParameters.CertificateValidator.Validate(x509SecurityKey.Certificate);
+                if (validationParameters.CertificateValidator == null)
+                    throw new NullReferenceException(ErrorMessages.IDX10232);
+
+                validationParameters.CertificateValidator.Validate(x509SecurityKey.Certificate);
+                return;
             }
+
+            throw new ArgumentOutOfRangeException("securityKey", ErrorMessages.IDX11009);
```

Comment:
I recommend changing this code flow around to make it easier to read and more maintainable.

`X509SecurityKey x509SecurityKey = securityKey as X509SecurityKey;`
`if (x509SecurityKey == null)`
`    throw new ArgumentOutOfRangeException("securityKey", ErrorMessages.IDX11009);`
`else `
`{`
`    ...`
`}`